library/src/main/java/com/novoda/downloadmanager/LiteDownloadManagerCommands.java

```diff
@@ -7,33 +7,113 @@
 
+    /**
+     * Retrieves all stored batches from the persistence layer and queues them ready for download,
+     * repopulating the internal map. Normally called when first starting the app, after a migration or restart.
+     *
+     * @param callback to be notified when all downloads are queued.
+     */
     void submitAllStoredDownloads(AllStoredDownloadsSubmittedCallback callback);
 
+    /**
+     * Retrieves a list of {@link DownloadBatchStatus} synchronously. Clients should
+     * specify their own Threading mechanism.
+     *
+     * @return a list of {@link DownloadBatchStatus}.
+     */
     @WorkerThread
     List<DownloadBatchStatus> getAllDownloadBatchStatuses();
 
+    /**
+     * Retrieves a list of {@link DownloadBatchStatus} passing to the {@link AllBatchStatusesCallback}.
+     *
+     * @param callback to pass the list of {@link DownloadBatchStatus} to.
+     */
     void getAllDownloadBatchStatuses(AllBatchStatusesCallback callback);
 
+    /**
+     * Retrieves {@link DownloadFileStatus} with the associated {@link DownloadBatchId} and {@link DownloadFileId}.
+     * Will return null if a {@link DownloadFileStatus} is not found with the matching {@link DownloadBatchId} and {@link DownloadFileId}.
+     * This call is synchronous, clients should specify their own Threading mechanism.
+     *
+     * @param downloadBatchId of the {@link DownloadFileStatus} to retrieve.
+     * @param downloadFileId  of the {@link DownloadFileStatus} to retrieve.
+     * @return {@link DownloadFileStatus} or null.
+     */
     @Nullable
     @WorkerThread
     DownloadFileStatus getDownloadFileStatusWithMatching(DownloadBatchId downloadBatchId, DownloadFileId downloadFileId);
 
+    /**
+     * Retrieves {@link DownloadFileStatus} with the associated {@link DownloadBatchId} and {@link DownloadFileId}.
+     * Will return null if a {@link DownloadFileStatus} is not found with the matching {@link DownloadBatchId} and {@link DownloadFileId}.
+     *
+     * @param downloadBatchId of the {@link DownloadFileStatus} to retrieve.
+     * @param downloadFileId  of the {@link DownloadFileStatus} to retrieve.
+     * @param callback        to pass the {@link DownloadFileStatus} to.
+     */
     void getDownloadFileStatusWithMatching(DownloadBatchId downloadBatchId, DownloadFileId downloadFileId, DownloadFileStatusCallback callback);
 
+    /**
+     * Changes the {@link ConnectionType} that must be satisfied in order to download.
+     * Will pause all current downloads that are violating this condition, resuming
+     * all those that satisfy the condition.
+     *
+     * @param allowedConnectionType that must be satisfied in order to download.
+     */
     void updateAllowedConnectionType(ConnectionType allowedConnectionType);
 
+    /**
+     * Adds a {@link CompletedDownloadBatch} to the download manager.
+     * This call is synchronous, clients should specify their own Threading mechanism.
```

Comment:
I would add something like, This is a long-running blocking operation, clients should call this method on their own Threading mechanism. Same for everywhere we use synchronous.

I'm mentioning this because that a method is synchronous is not a problem, the majority are. The problem is that this is a long running blocking operation.